extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/OtTracerPropagator.java

```diff
@@ -73,11 +73,10 @@ public static OtTracerPropagator getInstance() {
     // Baggage is only injected if there is a valid SpanContext
     Baggage baggage = Baggage.fromContext(context);
     if (!baggage.isEmpty()) {
+      // Metadata is not supported by OpenTracing
       baggage.forEach(
-          (key, value, metadata) -> {
-            // Metadata is not supported by OpenTracing
-            setter.set(carrier, PREFIX_BAGGAGE_HEADER + key, value);
-          });
+          (key, baggageEntry) ->
+              setter.set(carrier, PREFIX_BAGGAGE_HEADER + key, baggageEntry.getValue()));
```

Comment:
This seems to generally make iterating more tedious by requiring `getValue` - is it better for users? I think the talk about the generic interface is more theoretical and these actual usages are what would drive wanting to make a change.

Reply:
Without this you can look at the Builder and we force an extra allocation. I think just for that and it is still worth it.

Reply:
The builder's an implementation detail though - we can fix the allocation. So it's still more important to look at the iteration pattern itself - the previous seemed easier. It's not way easier though, so I wouldn't block a change but am not convinced it's much better this way either.

Reply:
For me this is a clear improvement:
* Reduces allocation (that can be done in other ways I agree)
* Reduces API surface by removing one extra interface (better for auto-instrumentation)
* Improves consistency with other "collection like" classes that we expose
* Downside: One extra call for the user in some cases.

Reply:
> Downside: One extra call for the user in some cases.

most cases :P I think generally we would prioritize the user experience among those bullets, so for me it's the hard to quantify value of consistency with other collection classes vs making it harder to access the baggage value. I also have an assumption that almost no users will be using metadata, which increases the negative impact of this change a bit. Still not a big deal but it's something so we should be sure.

Reply:
We talked today that users will not actually read baggage that much. They will write it more than read. And I slightly like @bogdandrutu 's change.